### PR TITLE
[FIX] web: command palette

### PR DESCRIPTION
**Pull request by FrancoisGe**
Before this commit, if a user does a search and press enter before the commands
are loaded, then the wrong command will be executed.
So we will wait for the commands to be loaded before executing the right command.

Solves the navigation problems with the arrows in the command palette on firefox.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/webclient/commands/command_palette.js [hunk 1]
```diff
@@ -237,6 +237,7 @@ export class CommandPalette extends Component {
 
     async executeSelectedCommand() {
         await this.inputPromise;
+        await this.keepLast;
```
Comment (brboi) [inline addons/web/static/src/webclient/commands/command_palette.js:240]: These two lines seem confusing and could use a bit of explanations

**Review by brboi — CHANGES_REQUESTED** [review]
Split the two fixes, add tests.

For the firefox problem: if testing is impossible, the commit message should be really explicit on what happened and how the fix is good.

File: addons/web/static/src/webclient/commands/command_palette.js (hunk 1 above)
Comment (aab-odoo) [reply addons/web/static/src/webclient/commands/command_palette.js:240]: Mostly, this line is useless, this isn't how keepLast works (keepLast isn't a promise, it's a class with an "add" function, that returns a promise

**Review by aab-odoo — COMMENTED** [review]
the first line of commit message would deserve a bit more explanation :-)
e.g. [FIX] web: concurrency issue in command palette

File: addons/web/static/tests/webclient/commands/command_palette_tests.js [hunk 2]
```diff
@@ -366,6 +367,46 @@ QUnit.test("multi provider with the same namespace", async (assert) => {
...
+    });
+    await nextTick();
+    assert.containsOnce(target, ".o_command_palette");
+    assert.containsN(target, ".o_command", 2);
+
+    await editSearchBar("b");
+    triggerHotkey("enter");
+    imSearchDef.resolve();
+    await nextTick();
+    assert.verifySteps(["b"]);
```
Comment (brboi) [inline addons/web/static/tests/webclient/commands/command_palette_tests.js:407]: I don't understand the test, ite seems wrong

I would have add an await nextTick() after the triggerHotkey and an empty verifySteps([]) before the imSearch.resolve()

File: addons/web/static/src/webclient/commands/command_palette.js [hunk 3]
```diff
@@ -212,7 +212,7 @@ export class CommandPalette extends Component {
 
         const listbox = this.el.querySelector(".o_command_palette_listbox");
         const command = listbox.querySelector(`#o_command_${nextIndex}`);
-        scrollTo(command.children[0], listbox);
+        scrollTo(command, listbox);
```
Comment (brboi) [inline addons/web/static/src/webclient/commands/command_palette.js:215]: The commit message does not mention this change

**Comment by aab-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**